app/main.py

```diff
@@ -59,30 +60,27 @@ def metrics(segments):
         ec = {}
 
     for n, d in G.degree():
-        nodes[n] = {
-            'degree': d,
-            'weightedDegree': wd[n],
-            'betweenness': bc[n],
-            'closeness': cc[n]
-        }
+        node_metrics = NodeInPlayMetrics(
+            degree=d,
+            weightedDegree=wd[n],
+            betweenness=bc[n],
+            closeness=cc[n],
+            eigenvector=None,
+        )
+        nodes[n] = node_metrics
         if n in ec:
-            nodes[n]['eigenvector'] = ec[n]
-
-    return {
-        'size': size,
-        'density': nx.density(G),
-        'diameter': max(path_lengths) if len(path_lengths) else 0,
-        'averagePathLength': (sum(path_lengths) / len(path_lengths))
-        if len(path_lengths) else 0,
-        'averageDegree': sum([d for n, d in G.degree()]) / size,
-        'averageClustering': nx.average_clustering(G),
-        'maxDegree': max_degree,
-        'maxDegreeIds': max_degree_ids,
-        'numConnectedComponents': nx.number_connected_components(G),
-        'numEdges': G.number_of_edges(),
-        'nodes': nodes
-    }
-
-
-if __name__ == '__main__':
-    metrics.interface.cli()
+            nodes[n].eigenvector = ec[n]
```

Comment:
I was unsure about this conditional setting of the eigenvector value.

I used eigenvector=None above. What was the output of the hug based API here? Did the nodeinplaymetrics simply not have an eigenvector key?

Is it supposed to be None, zero or not having the key?

FastAPI and its underlying Pydantic are very strict regarding this. If you want to init a NodeInPlayMetrics without an eigenvector value, it fails.

Reply:
Yes, the current implementation just omits the `eigenvector` property if the calculation fails. You can see this when posting the response from the above mentioned URL to the metrics service (assuming its running on localhost:8030):

```sh
http https://dracor.org/api/v1/corpora/rus/plays/lermontov-strannyj-chelovek | http :8030/metrics
```

To set it to `None`/`null` instead should be ok. I will check this with dracor-api after merging.